routing/rt/graph/graph.go

```diff
@@ -0,0 +1,536 @@
+// Copyright (c) 2016 Bitfury Group Limited
+// Distributed under the MIT software license, see the accompanying
+// file LICENSE or http://www.opensource.org/licenses/mit-license.php
+
+package graph
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math"
+	"math/rand"
+	"os"
+	"strconv"
+	"bytes"
+)
+
+var (
+	CyclicDataError         = errors.New("Cyclic data")
```

Comment:
It is an internal error. It is only used when creating explicit path after BFS. Maybe it should be renamed to InternalError. 
The graph implementation allows cyclic paths.